### DISCUSSION TITLE
Rename `priority_weight` attribute(s)

### DISCUSSION THREAD

**Discussion by michaelmicheal** [apache/airflow#21857] [General]
### Description

Currently, the priority weight attributes in the BaseOperator and TaskInstance classes have misleading names. 

As I understand it, BaseOperator has two priority weight attributes: `priority_weight` and `priority_weight_total`, where `priority_weight` is the user defined single-task priority weight, and `priority_weight_total` represents the total priority of the task and all downstream tasks based on a weight rule.

TaskInstance has a single priority weight attribute: `priority_weight`. However, when TaskInstance attributes are defined, `priority_weight` is defined as the task's  `priority_weight_total`.
```python
        self.priority_weight = task.priority_weight_total
```
This is potentially confusing since BaseOperator and TaskInstance both have the attribute `priority_weight`, but represent different measures of priority.

### Use case/motivation

I'm wondering if we can rename the `priority_weight` attribute in the BaseOperator class and/or the TaskInstance class to prevent confusion.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by SamWheating** [comment]
This can get especially misleading in the `Task Instance details` page where there `priority_weight` is listed twice (potentially with different values if there are downstream tasks)

**Comment by potiuk** [comment]
The `priority_weight_total` includes downstream or upstream tasks prioroties or both -depending on the weight rule. This is very clear to me. TaskInstance when executing is considered to have priority based on total weight (including upstram or downstram or both priorities). 

This is natural consequences that Task is a definition (static) and TaskInstance is "runtime" (i.e. it contains acual status used for scheduling). I see completely no problem with it. Not sure what the proposal is here. 

@michaelmicheal  - if you have a concrete proposal please open a PR and explain what you want to improve here.

I am also converting this into discussion.
